test/InputOutput/unit_spectralelement2d.jl

```diff
@@ -67,31 +63,86 @@ end
     Nq = 4
     quad = Quadratures.GLL{Nq}()
     mesh = Meshes.RectilinearMesh(domain, n1, n2)
+    grid_topology = Topologies.Topology2D(context, mesh)
+
+    return Spaces.SpectralElementSpace2D(
+        grid_topology,
+        quad;
+        enable_bubble,
+        enable_mask,
+    )
+end
+
+function mktempfile(f)
+    mktempdir() do dir
+        cd(dir) do
+            f(tempname(dir))
+        end
+    end
+end
+
+@testset "restart test for 2D spectral element simulations" begin
     device = ClimaComms.device()
     @info "Using device" device
-    context = ClimaComms.SingletonCommsContext(device)
-    grid_topology = Topologies.Topology2D(context, mesh)
+    context = ClimaComms.context(device)
```

Comment:
I think this test only works for SingletonCommsContext.

If you use MPI, `mktempfile` will create two different files on in the `/tmp/` of different nodes.

Reply:
This might be a really nice helper function in ClimaComms

Reply:
Can you suggest how we do this here, temporarily?

Reply:
Using temporary directories is a little tricky. If you create a directory with one MPI process, you have no guarantee that the other process sees the same directory (because it can take a little bit of time for the distributed filesystem to be synchronized across nodes)

What you can do instead is to create a temporary file in the current directory. This is not a problem because the creation of the file is handled by the HDF5+MPI library, which knows how to properly talk to the filesystem.

So, I think this would work
```julia
function mktempfile(f, context)
filename = ClimaComms.iamroot(context) ? tempname(pwd(); cleanup = true) : ""
filename = ClimaComms.bcast(context, filename)
f(filename)
end
```

Reply:
I applied the suggestion, seems to be working.